### ISSUE TITLE
Certificate verification issues

### ISSUE THREAD

**Issue by sambrightman** [melpa/melpa#5109]
I'm not sure if this is a misconfiguration of melpa.org or of various trust stores. I have not been able to figure out what is going on.

A recent version of development emacs (27.0.50) has had fallback to insecure `openssl s_client` removed and fails to verify the melpa.org certificate using gnutls-cli on Travis (Ubuntu Trusty container, `ca-certificates` up-to-date):

https://travis-ci.org/sambrightman/virtualenvwrapper.el/jobs/299906001

If I run the same command locally on on my Mac, I also get a failure:

```
- Status: The certificate is NOT trusted. The certificate issuer is unknown. The certificate chain uses insecure algorithm.
*** PKI verification of server certificate failed...
*** Fatal error: Error in the certificate.
```

I can also see that the old openssl command highlights a problem (but as a debug command, doesn't fail). Browser on the same machine seems to accept the certificate though. Is this thread:

https://community.letsencrypt.org/t/solved-why-isnt-my-certificate-trusted/2479/4

perhaps related?

**Comment by milkypostman** [comment]
i have seen this error recently but i don't understand what has changed. based on the forum post that you linked to our cert configuration for nginx is correct. i.e., we are, and according the post *should*, reference the `fullchain.pem`.

i cannot reproduce this anymore though.

**Comment by sambrightman** [comment]
Hmm. Did you read the later comments about a) Apache mis-behaving on certain versions; b) configuration being overriden with global values?

https://www.ssllabs.com/ssltest/analyze.html?d=melpa.org looks okay to  me.

I've noticed that supplying an incorrect path to the certificate file will not produce a warning/error about the file being missing, and instead assume an empty list of trusted CAs. However - from what I can tell - Emacs only attempts to use trust files which actually exist.

https://packages.ubuntu.com/trusty/all/ca-certificates/filelist doesn't seem to contain Let's Encrypt, even though it has been updated in June 2017. However, it does include DST Root CA X3.
